containers/Incentives/IncentiveCalculator.js

```diff
@@ -0,0 +1,190 @@
+                    benchmarksByProductId{
+                        nodes{
+                            benchmark
+                            eligibilityThreshold
+                        }
+                    }
+                }
+            }
+        }
+    `;
+
+const productsByBcghgidQuery = graphql`
+        query IncentiveCalculatorProductsByBcghgidQuery($bcghgidInput: String) {
+            getProductsByBcghgid(bcghgidInput: $bcghgidInput){
+                nodes{
+                    rowId
+                    quantity
+                    processingUnit
+                    applicationId
+                    units
+                    associatedEmissions
+                    attributableFuelPercentage
+                }
+            }
+        }
+    `;
+
+const carbonTaxByBcghgidQuery = graphql`
+    query IncentiveCalculatorCarbonTaxByBcghgidQuery($bcghgidInput: String, $reportingYear: String) {
+        getCarbonTaxByBcghgid(bcghgidInput:$bcghgidInput, reportingYear:$reportingYear){
+            nodes{
+                reportId
+                organisationId
+                fuelType
+                calculatedCarbonTax
+            }
+        }
+    }
+`;
+
+class IncentiveCalculatorContainer extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            totalCarbonIncentive:0,
+            incentiveSegments:[]
+        }
+    }
+
+    getData = async () => {
+      const allProducts = await fetchQuery(environment, allProductsQuery);
+      const reportedProducts = await fetchQuery(environment, productsByBcghgidQuery, {bcghgidInput:'4'});
+      const carbonTaxByBcghgid = await fetchQuery(environment, carbonTaxByBcghgidQuery , {bcghgidInput: '12111130401', reportingYear:'2014'});
+
+      const totalCarbonTax = carbonTaxByBcghgid.getCarbonTaxByBcghgid.nodes.reduce((total, curr) => {
+          return parseFloat(total) + parseFloat(curr.calculatedCarbonTax)
+      }, 0);
+
+      return ({
+          allProducts,
+          reportedProducts,
+          carbonTaxPaid: totalCarbonTax // assume a value until ciip and swrs are connected
+      });
+    };
+
+    generateIncentiveCalculationData = async () => {
+        const data = await this.getData();
+        console.log('data', data)
+        const productsReported = data.reportedProducts.getProductsByBcghgid.nodes;
+        const allProducts = data.allProducts.allProducts.nodes;
+        let totalCarbonIncentive = 0;
+        let incentiveSegments = [];
+
+        productsReported.forEach((product) => {
+           // Get bm/et details for the product from the Products table
+           const productDetails = allProducts.filter((p) => p.name === product.processingUnit);
+           const productQuantity = parseFloat(product.quantity)
+           const attributableFuelPercentage = parseFloat(product.attributableFuelPercentage)
+           const benchmark = productDetails[0].benchmarksByProductId.nodes[0].benchmark;
```

Comment:
@hamzajaved This may be more of an issue with how benchmarks are created / managed. But getting a Product's benchmark nodes and always selecting nodes[0] doesn't necessarily get the current benchmark for that Product.

example: a benchmark is created 1999-2001 it is node[0], a new benchmark is created 2001-2003 it is node[1] etc
Should we add a TODO here to document this issue?

Reply:
Yes, I believe this is a relic from before we added dates. will add a todo